pkgs/top-level/release-haskell.nix

```diff
@@ -215,6 +238,50 @@ let
         ;
 
       elmPackages.elm = pkgsPlatforms.elmPackages.elm;
+
+      # GHCs linked to musl.
+      pkgsMusl.haskell.compiler = packagePlatforms pkgs.pkgsMusl.haskell.compiler // {
+        # remove musl ghc865Binary since it is known to be broken and
+        # causes an evaluation error on darwin.
+        # TODO: remove ghc865Binary altogether and use ghc8102Binary
+        ghc865Binary = {};
+
+        # remove integer-simple because it appears to be broken with
+        # musl and non-static-linking.
+        integer-simple = {};
+      };
+
+      # Get some cache going for MUSL-enabled GHC.
+      pkgsMusl.haskellPackages =
+        removePlatforms
+          [
+            "aarch64-linux" # aarch64 does not appear to be supported
```

Comment:
Doesn't it just time out? I would have guessed it should work.

Reply:
Ah, yeah, I wasn't sure what was going on here.

With the current status of this PR, I removed this line:

```diff
diff --git a/pkgs/top-level/release-haskell.nix b/pkgs/top-level/release-haskell.nix
index 636f0f2b15e..14288b372cb 100644
--- a/pkgs/top-level/release-haskell.nix
+++ b/pkgs/top-level/release-haskell.nix
@@ -258,7 +258,7 @@ let
       pkgsMusl.haskellPackages =
         removePlatforms
           [
-            "aarch64-linux" # aarch64 does not appear to be supported
             "x86_64-darwin" # musl only supports linux
           ]
           {
```

I then reran `hydra-eval-jobs`:

```console
$ hydra-eval-jobs -I . pkgs/top-level/release-haskell.nix
```

Here's the diff comparing running `hydra-eval-jobs` on `release-haskell.nix` with the current status of this PR vs. making the above change:

```diff
--- current-state    2021-07-24 14:28:55.013105845 +0900
+++ building-on-aarch64    2021-07-24 15:37:46.510185699 +0900
@@ -345373,6 +345373,9 @@
     "system": "x86_64-linux",
     "timeout": 36000
   },
+  "pkgsMusl.haskellPackages.hello.aarch64-linux": {
+    "error": "error: Failed to evaluate ghc-8.10.4: «unsupported»: is not supported on ‘aarch64-linux’"
+  },
   "pkgsMusl.haskellPackages.hello.x86_64-linux": {
     "description": "Hello World, an example package",
     "drvPath": "/nix/store/15dsk39ajxxk71fvp92jjp3cz56qafj8-hello-1.0.0.2.drv",
@@ -345390,6 +345393,9 @@
     "system": "x86_64-linux",
     "timeout": 36000
   },
+  "pkgsMusl.haskellPackages.lens.aarch64-linux": {
+    "error": "error: Failed to evaluate ghc-8.10.4: «unsupported»: is not supported on ‘aarch64-linux’"
+  },
   "pkgsMusl.haskellPackages.lens.x86_64-linux": {
     "description": "Lenses, Folds and Traversals",
     "drvPath": "/nix/store/5yxm6fhlaiq6lw29aypzx5rvir6qirq4-lens-4.19.2.drv",
@@ -345407,6 +345413,9 @@
     "system": "x86_64-linux",
     "timeout": 36000
   },
+  "pkgsMusl.haskellPackages.random.aarch64-linux": {
+    "error": "error: Failed to evaluate ghc-8.10.4: «unsupported»: is not supported on ‘aarch64-linux’"
+  },
   "pkgsMusl.haskellPackages.random.x86_64-linux": {
     "description": "Pseudo-random number generation",
     "drvPath": "/nix/store/k7fhjmvf6ygqm3izzc0i94d2j2rzxib0-random-1.2.0.drv",
```

Reply:
Oh yeah, I see why this happens: We can only natively bootstrap GHC with musl on x86_64-linux because we only have a musl bindist for this platform since upstream doesn't provide musl builds for aarch64.

With this in mind you probably should just clarify the comment and keep it as is.

Reply:
Fixed this up in 95cd2f7a1f7cedaf4fe.